[Feature] main site navigation

yonikid15 opened the pull request:
🤖 Resolves #10793 

## 👋 Introduction

- Create new nav menu for entire site
- Replaces admin side menu 

## 🕵️ Details

- Ran into a bug after cleaning up my code, so I pushed up what I had before that
- I noticed that the role switcher is going to need some tweaking. Right now, it takes a few renders before the current nav role updates (which updates the nav menu view).
- I feel like there's a some styling things I missed as well. 
- I need to (possibly) add localized constants for the Role enum for the role sub menu. However, right now the display name for the roles comes from the rolePermission.php so maybe we can alter that. 

## 🧪 Testing

1. Play around with nav menu

petertgiles commented on packages/ui/src/components/NavMenu/NavMenu.tsx:
Clearly you know about this but just adding a discussion spot: the mobile menu doesn't work well when it's taller than the viewport.
![image](https://github.com/user-attachments/assets/95b337ef-fb0e-4c22-9df0-9d32611100f4)

yonikid15 replied:
This should be fixed now [caeb424](https://github.com/GCTC-NTGC/gc-digital-talent/pull/11501/commits/caeb424a0873a087ecf98d0a14e7f2a2323c0e51). Sorry my commits got messy at the end :sweat_smile:

petertgiles reviewed (changes requested):
This is really starting to look great.  :star_struck: I've done a quick readthrough and played with it a bit.  Here are some observations.

petertgiles commented on apps/web/src/components/Layout/AdminLayout/AdminLayout.tsx:
If it's the latter, maybe we can still use a single component and use it in two different places in the routing tree.

yonikid15 replied:
Consolidated the layouts [76c9d98](https://github.com/GCTC-NTGC/gc-digital-talent/pull/11501/commits/76c9d98fb4c1545cd66e11b9e0ba8eaaf70ad4cd)

petertgiles replied:
Yeah, keeping AdminLayout as a skeleton is a good in-between step.  :+1:

yonikid15 commented on packages/ui/src/components/NavMenu/NavMenu.tsx:
I haven't experienced on any browsers. Could you check again and let me know if it's still happening for you? If it is we could huddle and take a closer look.

yonikid15 commented on packages/ui/src/components/NavMenu/NavMenu.tsx:
Yes :+1:  I split the Community dashboard onto it's own page as we talked about previously.

yonikid15 commented on packages/ui/src/components/NavMenu/NavMenu.tsx:
I should have added the remaining links for these now

petertgiles replied:
Still got a few broken ones.
![image](https://github.com/user-attachments/assets/2da76e37-ea95-4084-bb30-f9c26bfe8078)
![image](https://github.com/user-attachments/assets/70fe58c7-fe71-4ec3-ae75-e7a64359aff5)

yonikid15 replied:
Added in [54d20e6](https://github.com/GCTC-NTGC/gc-digital-talent/pull/11501/commits/54d20e6705421d3b3cb4ed1e122e48e768a52bed)

yonikid15 commented on packages/ui/src/components/NavMenu/NavMenu.tsx:
Well the navRole context automatically drops to the least privileged role (in this case applicant). The easy solution would be create an account settings page for each role. Another would be saving the "prevNavRole" in the local state for account settings page, and using that for the MainNavMenu.

petertgiles replied:
The reason I suggested adding all the higher roles is because the nav role logic will prefer to keep the current nav role, instead of dropping, if the current nav role is allowed for the page.  This would avoid creating a copied page or adding extra controls to the context.

yonikid15 commented on packages/ui/src/components/NavMenu/NavMenu.tsx:
Nice catch! Fixed in [d4d0a0c](https://github.com/GCTC-NTGC/gc-digital-talent/pull/11501/commits/d4d0a0cb163bb8d9c506b0861cff6f6424053a17)

petertgiles reviewed (changes requested):
I just did another code readthrough and flagged a few more things.

petertgiles commented on api/lang/en/role.php:
You have a case statement in the frontend to get the names of these roles in a few different places.  I think making this enum localized is the way to go.

yonikid15 replied:
Moved localised messages into it's own file  [9ee5090](https://github.com/GCTC-NTGC/gc-digital-talent/pull/11501/commits/9ee50909442827d16aca5bde3103ff2fc024840a)

yonikid15 commented on api/app/Enums/Role.php:
```diff
@@ -6,6 +6,7 @@ enum Role: string
 {
     case BASE_USER = 'base_user';
     case APPLICANT = 'applicant';
+    case Manager = 'manager';
```
Fixed by [5a84068](https://github.com/GCTC-NTGC/gc-digital-talent/pull/11501/commits/5a84068ddf241b322d0e61727a9339567035994a)

yonikid15 commented on apps/web/src/components/NavMenu/MainNavMenu.tsx:
```diff
@@ -0,0 +1,228 @@
+
+import NotificationDialog from "../NotificationDialog/NotificationDialog";
+import useNavContext from "../NavContext/useNavContext";
+import { useMainLinks } from "./navlinks";
```
Renamed in [a97ab34](https://github.com/GCTC-NTGC/gc-digital-talent/pull/11501/commits/a97ab3485990f544821e5cf1944bede058cd2287)

yonikid15 commented on apps/web/src/components/NavMenu/navlinks.tsx:
```diff
@@ -0,0 +1,326 @@
+ * @param loggedIn If the user is logged in
+ * @returns
+ */
+export const useMainLinks = (
```
Renamed in [a97ab34](https://github.com/GCTC-NTGC/gc-digital-talent/pull/11501/commits/a97ab3485990f544821e5cf1944bede058cd2287)

yonikid15 commented on apps/web/src/components/Router.tsx:
```diff
@@ -44,6 +49,28 @@ const createRoute = (locale: Locales) =>
+              children: [
+                {
+                  index: true,
+                  // loader: async () => {
```
Removed loader in [6bd1b87](https://github.com/GCTC-NTGC/gc-digital-talent/pull/11501/commits/6bd1b8723b21ec5a0589807f0d2b7e31d24c58b2)

yonikid15 commented on apps/web/src/pages/AdminDashboardPage/AdminDashboardPage.tsx:
```diff
@@ -265,13 +265,13 @@ const DashboardPageApi = () => {
-      ROLE_NAME.PoolOperator,
-      ROLE_NAME.RequestResponder,
-      ROLE_NAME.CommunityManager,
+      // ROLE_NAME.PoolOperator,
```
Remove comments in [2509760](https://github.com/GCTC-NTGC/gc-digital-talent/pull/11501/commits/2509760fc32605fdced602a654ca22cdbe7d6adb)

yonikid15 commented on packages/ui/src/components/NavMenu/NavMenuWrapper.tsx:
```diff
@@ -0,0 +1,171 @@
+              }}
+            >
+              {/* <RemoveScroll
+                  enabled={isSmallScreen && open}
```
Removed in [90bfafc](https://github.com/GCTC-NTGC/gc-digital-talent/pull/11501/commits/90bfafc9d807d5ac47898f41cdb5d2fccbd47d3c)

yonikid15 commented on apps/web/src/components/NavMenu/MainNavMenu.tsx:
```diff
@@ -0,0 +1,228 @@
+  const { userAuthInfo } = useAuthorization();
+  const { loggedIn } = useAuthentication();
+
+  const [prevNavRole, setPrevNavRole] = useLocalStorage<NavRole | null>(
```
Reverted in [7d84bc3](https://github.com/GCTC-NTGC/gc-digital-talent/pull/11501/commits/7d84bc326acd18d9812aa6763ba7bcb2567f4cc6)

Agreed! I made some changes directly in the context that should keep things stable.

yonikid15 commented on packages/ui/src/components/NavMenu/NavMenu.tsx:
```diff
@@ -0,0 +1,249 @@
+          "data-h2-color":
+            "base(secondary.darker) base:dark(secondary.lightest) l-tablet:all(secondary.lighter)",
+          "data-h2-text-decoration":
+            "base(none) base:children[>span](none!important)", // TODO: FIX UNDERLINE WHEN ACTIVE
```
Removed in [738d62b](https://github.com/GCTC-NTGC/gc-digital-talent/pull/11501/commits/738d62b0e4f4ad9854d23a022b3753527bd06ffb)

yonikid15 commented on apps/web/src/components/NavMenu/navlinks.tsx:
```diff
@@ -0,0 +1,326 @@
+      (roleAssignment) => roleAssignment.role?.name !== ROLE_NAME.BaseUser,
+    )
+    .map((roleAssignment) => {
+      const role = convertRoleToNavRole(roleAssignment.role?.name as RoleName);
```
Fixed in [9ee5090](https://github.com/GCTC-NTGC/gc-digital-talent/pull/11501/commits/9ee50909442827d16aca5bde3103ff2fc024840a)

yonikid15 commented on apps/web/src/components/NavMenu/navlinks.tsx:
```diff
@@ -0,0 +1,326 @@
+    </SignOutConfirmation>
+  );
+
+  const getRoleName = {
```
Localised in [9ee5090](https://github.com/GCTC-NTGC/gc-digital-talent/pull/11501/commits/9ee50909442827d16aca5bde3103ff2fc024840a)

yonikid15 commented on apps/web/src/components/NavMenu/MainNavMenu.tsx:
```diff
@@ -0,0 +1,228 @@
+      (roleAssignment) => roleAssignment.role?.name !== ROLE_NAME.BaseUser,
+    );
+
+  const getRoleName: Record<string, string> = {
```
Localised in [9ee5090](https://github.com/GCTC-NTGC/gc-digital-talent/pull/11501/commits/9ee50909442827d16aca5bde3103ff2fc024840a)

yonikid15 commented on apps/web/src/components/NavMenu/MainNavMenu.tsx:
```diff
@@ -0,0 +1,228 @@
+
+  const [isNotificationDialogOpen, setNotificationDialogOpen] = useState(false);
+
+  const [navRoleState, setNavRoleState] = useState(navRole);
```
Removed in [d69b8c5](https://github.com/GCTC-NTGC/gc-digital-talent/pull/11501/commits/d69b8c52f7258462ffce42b4db9e6e4f871d8d68)

yonikid15 commented:
Okay so I think I fixed the nav context issue by storing the nav role to local storage, and then removing it once the user logs out. [359e25d](https://github.com/GCTC-NTGC/gc-digital-talent/pull/11501/commits/359e25d07c2ff9781642b9b92c03c894842bef86) and [a1b3c03](https://github.com/GCTC-NTGC/gc-digital-talent/pull/11501/commits/a1b3c03871c3d205b91c08073490791debd051d3)

petertgiles commented on packages/auth/src/const.ts:
```diff
@@ -11,15 +11,17 @@ export const ROLE_NAME = {
 type ObjectValues<T> = T[keyof T];
 export type RoleName = ObjectValues<typeof ROLE_NAME>;
+
+export const NAV_ROLE = "nav_role";
```
A bit more descriptive:
```suggestion
export const NAV_ROLE_STORAGE_KEY = "nav_role";
```

yonikid15 replied:
Nice call :+1: Fixed in [c0a75d8](https://github.com/GCTC-NTGC/gc-digital-talent/pull/11501/commits/c0a75d801f67b24a7a4ba7e113ff1ee664c2e0bd)

petertgiles commented on apps/web/src/components/NavContext/NavContextContainer.tsx:
```diff
@@ -1,8 +1,10 @@
-import { useLogger } from "@gc-digital-talent/logger";
+import { useLocalStorage } from "@gc-digital-talent/storage";
+
+import { NAV_ROLE } from "../../../../../packages/auth/src/const";
```
```suggestion
import { NAV_ROLE } from "@gc-digital-talent/auth/const";
```

yonikid15 replied:
Fixed in [c0a75d8](https://github.com/GCTC-NTGC/gc-digital-talent/pull/11501/commits/c0a75d801f67b24a7a4ba7e113ff1ee664c2e0bd)

petertgiles commented on apps/web/src/components/NavContext/NavContextContainer.tsx:
```diff
@@ -27,7 +29,7 @@ export const NavContext = createContext<NavContextState>({
-function convertRoleToNavRole(role: RoleName): NavRole {
+export function convertRoleToNavRole(role: RoleName): NavRole {
   switch (role) {
     case "guest":
```
Weren't we going to get rid of guest?

yonikid15 replied:
Removed guest role and replaced with null [61678be](https://github.com/GCTC-NTGC/gc-digital-talent/pull/11501/commits/61678bebb23697a9541d5ca1633fef9acd679735)

petertgiles commented on apps/web/src/components/NavMenu/MainNavMenu.tsx:
```diff
@@ -0,0 +1,203 @@
+      (roleAssignment) => roleAssignment.role?.name !== ROLE_NAME.BaseUser,
+    );
+
+  const roleNames: Record<string, string> = {
```
You can simplify this a bit and tighten the typing at the same time.  You'll find a gap with "guest", too, if you do this!
![image](https://github.com/user-attachments/assets/6fd22903-9464-4cee-af76-96af10be2a5a)

petertgiles replied:
Same recommendation for the maps in apps/web/src/components/NavMenu/useMainNavLinks.tsx

yonikid15 replied:
I was able to change [88c420b](https://github.com/GCTC-NTGC/gc-digital-talent/pull/11501/commits/88c420b6d4e479bcff054605bf700eea3809dbe5) the above. However, I'm not sure how it would work in `useMainNavLinks.tsx`, since the `role.name` returns a string in camal_case so the mapping doesn't work out.

![image](https://github.com/user-attachments/assets/567361b5-f49d-4a76-8370-f7d60d5595a4)

petertgiles replied:
You have to quote the property name, but it doesn't mind snake case.
![image](https://github.com/user-attachments/assets/2c98c422-af16-4f8e-802a-60d1a8e643d3)

yonikid15 replied:
This would work, but we set our eslint rules to automatically remove the quotes on object keys when saving :laughing: I'm not sure where to go from here.

petertgiles commented on apps/web/src/components/NavMenu/useMainNavLinks.tsx:
```diff
@@ -0,0 +1,330 @@
+      (roleAssignment) => roleAssignment.role?.name !== ROLE_NAME.BaseUser,
+    )
+    .map((roleAssignment) => {
+      const role = roleAssignment.role?.name ?? "";
```
I'd recommend just testing for falsy here and return early instead of putting junk rows in your map objects.

yonikid15 replied:
Yea that's better :+1: [1ea6670](https://github.com/GCTC-NTGC/gc-digital-talent/pull/11501/commits/1ea6670ccf30682052a5c179128a2355c047bffe)

petertgiles commented on apps/web/src/components/NavMenu/useMainNavLinks.tsx:
```diff
@@ -0,0 +1,330 @@
+        mainLinks: [
+          Home,
+          BrowseJobs,
+          // // uncomment to test scrolling on mobile view
```
Can probably get rid of this now.

yonikid15 replied:
Removed in [1ea6670](https://github.com/GCTC-NTGC/gc-digital-talent/pull/11501/commits/1ea6670ccf30682052a5c179128a2355c047bffe)

petertgiles reviewed (changes requested):
Funcationality looks great!

Just a few more small code things.  Let me know if you want to get on a call to discuss and get cleared up quickly.

tristan-orourke reviewed (changes requested):
Played around with the branch locally. Here are my general notes about the context switcher:
- Role menu shouldn't appear at all before logging in
- Role menu also shouldn't appear if user only has one Role (ie is an Applicant)
- Going to account settings switches always switches the nav to Applicant context. The intention of #11034 was that any role could go to the account settings page without losing their nav context
- Switching of contexts isn't reflected in browser history. eg If I go through these pages: `Community Dashboard > (Community) Processes > Admin Dashboard > (Admin) Processes`, then use `Back` twice, I end up on `(Admin) Processes` instead of `(Community) Processes`. This is definitely my own oversight from planning this out, but it annoys me now that I've realized it. Maybe it would be better to store the context in browser state than in localStorage? But its also an low-importance edge case, and changing this could be pushed to its own ticket.

Here are some specific issues I noticed regarding pages or permissions, but I'm not sure if they really belong to this ticket:
- Users and Processes can be viewed in Admin or Community context. Requests also appears in Admin nav, but always switches you to Community context.
 - Community Admin sees Candidates in nav, but doesn't have access permission
 - Manager dropdown links to Manager "Home" instead of Dashboard, and Dashboard is a 404. Is that because the Manager Dashboard page will be made in another ticket?

vd1992 commented on apps/web/src/components/NavMenu/useMainNavLinks.tsx:
Contrast accessibility issue on the active page highlight

![image](https://github.com/user-attachments/assets/0705731f-ef8f-4f0f-a718-fd9619ebf177)

yonikid15 replied:
Fixed in [1088229](https://github.com/GCTC-NTGC/gc-digital-talent/pull/11501/commits/108822941d0edb498a27691ae7df0286b8306a21)

vd1992 commented on apps/web/src/pages/CommunityDashboardPage/CommunityDashboardPage.tsx:
Not sure if the stuff below is in scope, but I noticed

Logged in as Process Operator, what is supposed to go in the gap?

![image](https://github.com/user-attachments/assets/31a72e73-d348-4d18-9354-683086e544ed)

Community Admin can't reach community pages

![image](https://github.com/user-attachments/assets/7a463470-ef9e-496e-ad3c-791b7e1a0f48)

yonikid15 replied:
This is how it is currently on the main branch and is out if scope for this issue. Pretty sure there's an issue coming soon for creating a community admin dashboard.

vd1992 reviewed (commented):
Just poking around a bit to help things along, looks neat. Found some stuff to bring to attention below

yonikid15 commented on apps/web/src/components/NavMenu/MainNavMenu.tsx:
Fixed in [1088229](https://github.com/GCTC-NTGC/gc-digital-talent/pull/11501/commits/108822941d0edb498a27691ae7df0286b8306a21)

yonikid15 commented on apps/web/src/components/NotificationDialog/NotificationDialog.tsx:
Fixed in [c732892](https://github.com/GCTC-NTGC/gc-digital-talent/pull/11501/commits/c732892073f7fc83b6e382c00a55b4bb16353d0e)

tristan-orourke reviewed (changes requested):
The main issues from my previous look are resolved! 👍 However, I've noticed a few more things.

The Community and Admin contexts show fewer links in the nav than users currently see in the side nav on the admin site. I think people are going to miss things like links to the skills list, and now job templates. Let's check-in with @JoshBeveridge again on the current set of links.

Breadcrumbs on non-applicant pages can be a bit weird. First, "Home" on those breadcrumbs leads back to the Dashboard, instead of Home. This is especially confusing because we _do_ have a real link "Home" in the nav bar now. Also, this home link leads back to the "Admin" dashboard, even if in Community context. This leads to an authentication error for non-platform admins!

Logging in as legacy-community@test.com (which has Community Manager role, like our current manager users) seems to totally break the nav 😖 .
![image](https://github.com/user-attachments/assets/1e73cff6-9909-4d6d-a1cc-5e0e7b5d134e)
I could only log out by manually visiting http://localhost:8000/en/logged-out.

A Request Responder (request@test.com) sees Processes link in nav bar, but doesn't have access to that page. A user should not see nav links to a page they don't have permission to access.

Similarly, a Platform Admin without other roles sees Requests in the nav bar, but isn't authorized to view it.

One more minor thing, but its odd: if I enter http://localhost:8000/en/ directly in the browser url bar, I will be on the home page, but the Home item on the nav isn't highlighted. This seems a little inconsistent - sometimes it works? Not sure what's going on there.

yonikid15 commented:
@tristan-orourke made some fixes
> The Community and Admin contexts show fewer links in the nav than users currently see in the side nav on the admin site. I think people are going to miss things like links to the skills list, and now job templates. Let's check-in with @JoshBeveridge again on the current set of links.

Sounds good :+1: I'll leave a question in the channel.
> 
> Breadcrumbs on non-applicant pages can be a bit weird. First, "Home" on those breadcrumbs leads back to the Dashboard, instead of Home. This is especially confusing because we _do_ have a real link "Home" in the nav bar now. Also, this home link leads back to the "Admin" dashboard, even if in Community context. This leads to an authentication error for non-platform admins!
>

I was able to fix this by setting the `homePath` in the `useBreadcrumbs` hook.  [423bb95](https://github.com/GCTC-NTGC/gc-digital-talent/pull/11501/commits/423bb95a5eb1a9ee20784985f682bca7f53f9a5a)

> Logging in as [legacy-community@test.com](mailto:legacy-community@test.com) (which has Community Manager role, like our current manager users) seems to totally break the nav 😖 . I could only log out by manually visiting http://localhost:8000/en/logged-out.

Hmm, I wasn't able to reproduce this. If I had to guess your logged in, but the `nav_role` in locale storage didn't get picked up. Could you try checking the locale storage when this happens to see if the `nav_role` key exists. I could join you in a huddle as well.

> 
> A Request Responder ([request@test.com](mailto:request@test.com)) sees Processes link in nav bar, but doesn't have access to that page. A user should not see nav links to a page they don't have permission to access.

> Similarly, a Platform Admin without other roles sees Requests in the nav bar, but isn't authorized to view it.

This fix should only show links in the nav menu that are viewable by role. [63b729f](https://github.com/GCTC-NTGC/gc-digital-talent/pull/11501/commits/63b729fdb3b7807c6b8147ec7e2b077dafc1dc0d)

> One more minor thing, but its odd: if I enter http://localhost:8000/en/ directly in the browser url bar, I will be on the home page, but the Home item on the nav isn't highlighted. This seems a little inconsistent - sometimes it works? Not sure what's going on there.

This is a pre-existing bug, I'll make a issue for it. You can visit https://talent.canada.ca/en/ and you'll see it's happening on the live site as well.

tristan-orourke commented:
> > Breadcrumbs...
>
> I was able to fix this by setting the `homePath` in the `useBreadcrumbs` hook. [423bb95](https://github.com/GCTC-NTGC/gc-digital-talent/pull/11501/commits/423bb95a5eb1a9ee20784985f682bca7f53f9a5a)

Cool! I like that our breadcrumbs are already somewhat "smart". We might need to do more with this soon. Jerbo was talking about having breadcrumbs reflect how you get to a page, not just what page you're on.
 
> > Issues with legacy-community@test.com...
> 
> Hmm, I wasn't able to reproduce this. If I had to guess your logged in, but the `nav_role` in locale storage didn't get picked up. Could you try checking the locale storage when this happens to see if the `nav_role` key exists. I could join you in a huddle as well.

Thanks. Yeah, this looks fine now for me as well.

> > Links to pages you can't access...
> 
> This fix should only show links in the nav menu that are viewable by role. [63b729f](https://github.com/GCTC-NTGC/gc-digital-talent/pull/11501/commits/63b729fdb3b7807c6b8147ec7e2b077dafc1dc0d)

That's great, but I have an additional suggestion now. We're duplicating these lists of roles now in a couple places. At least here, in the nav link, and on the pages themselves (passed in to RequireAuth component). Could we extract these arrays to a common constants file, which has an array of "allowed" roles for each page, to ensure these are in sync?

> > One more minor thing, but its odd: if I enter http://localhost:8000/en/ directly in the browser url bar, I will be on the home page, but the Home item on the nav isn't highlighted. This seems a little inconsistent - sometimes it works? Not sure what's going on there.
> 
> This is a pre-existing bug, I'll make a issue for it. You can visit https://talent.canada.ca/en/ and you'll see it's happening on the live site as well.

Got it 👍 . Please make that issue.

tristan-orourke reviewed (changes requested):
Please just ensure that the nav buttons and the page themselves are referencing the roles which are allowed to view them from a common place.

yonikid15 commented:
> Please just ensure that the nav buttons and the page themselves are referencing the roles which are allowed to view them from a common place.

Okay I moved the permission constants into a hook, and also added the new resources and system settings submenus

[99454a6](https://github.com/GCTC-NTGC/gc-digital-talent/pull/11501/commits/99454a6c08bb2d57607af2abbabbca30051b240e)

yonikid15 commented:
Josh Design Review:

- [ ] Image 1 - Contrast on the hover here definitely doesn't pass. Maybe force it to secondary.darkest on dark mode?
- [ ] Image 2 - Things don't quite seem center aligned to me - I think it's the divider line in particular?
- [ ] Can we make the dividers in the nav base:all(black.light)?
- [ ] Image 3 - The chevron looks different... is it because we're using the mini Heroicon?
- [ ] Image 4 - Dropdown padding is a bit off
- [ ] Image 5 - Hover on the notification icon is weird and also contrast again
- [ ] Image 6 - This width isn't ideal on admin, but assuming we can't really fix it right now, can we at least vertically align everything to the middle and have the dividers grow in a similar way to the one next to the notification bell?
- [ ] Image 7 - Can we ensure the mode switcher is centered with the menu? This wasn't in the design because I had all three elements in this section centered together rather than split apart, but I prefer the split you have, so centering makes more sense
- [ ] Image 8 - Right now you can open the notification panel on top of the menu, I think opening either should close the other if its open
- [ ] Image 9 - The page background for the notification pane is different from the menu/dialogs - we should make it the same (use the menu/dialog one)
- [ ] Image 10 - Can't remember if I mentioned this in my initial review, but the color of these buttons makes me nervous in dark mode... Maybe try inverting them to white?
- [ ] Image 11 - Can we adjust the design of the mobile menu card slightly? Ditch the external border and ensure the internal ones are base(black.darkest.2) base:dark(black.darkest.5)
- [ ] Image 12 - Contrast on some? of the menu items isn't swapping to the light value in dark mode
- [ ]  I still can't seem to open submenus with the arrow keys :thinking_face:

Image 1:
![image](https://github.com/user-attachments/assets/2d47eac0-8ac3-418b-99a0-20b216a5928b)

Image 2:
![image](https://github.com/user-attachments/assets/ae32708c-2861-4548-940e-96b5e2b9d054)

Image 3:
![image](https://github.com/user-attachments/assets/5c92ef54-0905-4f7e-ab03-35d9cb60cd2e)

Image 4:
![image](https://github.com/user-attachments/assets/0797c33d-806a-4537-804f-eff4b2b23adf)

Image 5:
![image](https://github.com/user-attachments/assets/c8d29d1e-0f3e-4850-86f5-602fd5ca56a2)

Image 6:
![image](https://github.com/user-attachments/assets/ff2d3202-44a0-419b-9d39-28d08ec46652)

Image 7:
![image](https://github.com/user-attachments/assets/72692594-478f-4d88-9e73-d601f0173231)

Image 8:
![image](https://github.com/user-attachments/assets/df58f548-f79f-46dc-b7df-21ca531264f2)

Image 9:
![image](https://github.com/user-attachments/assets/0850d506-6214-41cf-a615-0cf2418ef47c)

Image 10:
![image](https://github.com/user-attachments/assets/a25ba812-82ae-4755-ac5b-9bcfea37c650)

Image 11:
![image](https://github.com/user-attachments/assets/1f428119-bc65-4304-9ff5-50d286dd8c1f)

Image 12:
![image](https://github.com/user-attachments/assets/691a5e3c-7e7d-4cce-9761-ca4a30104c76)

tristan-orourke commented on apps/web/src/hooks/usePermissionConstants.ts:
```diff
@@ -0,0 +1,34 @@
+import { RoleName } from "@gc-digital-talent/auth";
+
+const usePermissionConstants = () => {
```
Actually, I don't think this should be a "hook". Putting code in a "hook" allows it to use some React magic, like other hooks, but comes with a downside: hooks can only be used inside React components. This is a simple set of constants, it doesn't reference any other hooks, so there's no reason to call it a hook and restrict it to React components.

I would move/rename this file to `apps/web/src/constants/permissionConstants.ts`

yonikid15 replied:
Sounds good, fixed in [654171e](https://github.com/GCTC-NTGC/gc-digital-talent/pull/11501/commits/654171e44414d2061a017efb530ea695de63d4b9)

petertgiles commented:
@yonikid15 Could you please check how the new department pages with the new hero look with the new nav?  Could you check if you need to add the extra x3 of top padding back in that I removed in [this commit](https://github.com/GCTC-NTGC/gc-digital-talent/pull/11853/commits/7db6149a2ba4163bf270d0abe74a9b61c0edd4de)?

yonikid15 commented:
> @yonikid15 Could you please check how the new department pages with the new hero look with the new nav? Could you check if you need to add the extra x3 of top padding back in that I removed in [this commit](https://github.com/GCTC-NTGC/gc-digital-talent/pull/11853/commits/7db6149a2ba4163bf270d0abe74a9b61c0edd4de)?

![image](https://github.com/user-attachments/assets/1615f987-daa4-4e35-9c46-300c5e1d56df)

Looks good!

petertgiles commented:
Wow, this looks incredible!  I can't wait to get it merged in.  The TOC menus look weird in Chromatic but they are behaving fine in the app so I won't worry about it.  :shrug: I noticed two small things.  The role menu seems to be sitting a few pixels higher than the rest of the menu items and the contrast is low on the hover effect on the buttons in light mode.  The hover looks great in dark mode.
![image](https://github.com/user-attachments/assets/bc19efb2-f4e0-4fa7-a0f7-2ee07291af9b)
![image](https://github.com/user-attachments/assets/80af5332-3a4c-48e2-a6ad-7c42250dff3b)

yonikid15 commented:
The color contrast passes the AA standards on contrast checker so we should be good on that!
Nice catch on the role sub menu vertical alignment :eagle: :eye:. Should be fixed in [18e3777](https://github.com/GCTC-NTGC/gc-digital-talent/pull/11501/commits/18e3777a538cc4041a574234dec532ffbf1bf84a)

petertgiles reviewed (approved):
Incredible!  It turned out so amazing.
